project/src/main/ui/chat/chat-library.gd

```diff
@@ -18,36 +18,36 @@ Parameters:
 	'level_num': (Optional) The current level being chosen; '1' being the creature's first level. If specified, this
 			allows the creature to say something about the upcoming level.
 """
-func load_chat_events_for_creature(creature: Creature, level_num: int = -1, chit_chat: bool = false) -> ChatTree:
+func load_chat_events_for_creature(creature: Creature, level_num: int = -1) -> ChatTree:
 	var state := {
 		"creature_id": creature.creature_id,
-		"notable_chat": PlayerData.chat_history.get_filler_count(creature.creature_id) > 0
+		"notable_chat": PlayerData.chat_history.get_filler_count(creature.creature_id) > 0,
+		"level_num": level_num
 	}
-	if level_num != -1:
-		state["level_num"] = level_num
+	
+	if state["level_num"] == -1:
+		state["level_num"] = _first_unfinished_level_num(creature)
+	
+	# returning dialog for the creature
+	var filler_ids := _filler_ids_for_creature(creature)
+	var chosen_dialog := choose_dialog_from_chat_selectors(creature.chat_selectors, state, filler_ids)
 	
 	var chat_tree: ChatTree
-	if chit_chat or level_num != -1 or not creature.get_level_ids():
-		var filler_ids := _filler_ids_for_creature(creature)
-		
-		# returning dialog for the creature
-		var chosen_dialog := choose_dialog_from_chat_selectors(creature.chat_selectors, state, filler_ids)
-		if creature.dialog.has(chosen_dialog):
-			chat_tree = ChatTree.new()
-			chat_tree.from_json_dict(creature.dialog[chosen_dialog])
-			chat_tree.history_key = "dialog/%s/%s" % [creature.creature_id, chosen_dialog]
-		else:
-			var path := "res://assets/main/creatures/primary/%s/%s.json" % \
-					[creature.creature_id, chosen_dialog.replace("_", "-")]
-			chat_tree = load_chat_events_from_file(path)
+	if creature.dialog.has(chosen_dialog):
+		chat_tree = ChatTree.new()
+		chat_tree.from_json_dict(creature.dialog[chosen_dialog])
+		chat_tree.history_key = "dialog/%s/%s" % [creature.creature_id, chosen_dialog]
 	else:
-		# returning a tree where the player can select creature dialog
+		var path := "res://assets/main/creatures/primary/%s/%s.json" % \
+				[creature.creature_id, chosen_dialog.replace("_", "-")]
+		chat_tree = load_chat_events_from_file(path)
+	
+	# set the launched_level property, so that levels will launch after the dialog completes
+	if state["level_num"] != -1:
```

Comment:
The old logic had `level_num >= 1` which is a little more bulletproof

Reply:
Changed to `>= 1`. Testing this code also helped me catch another bug which I fixed, where some code was referencing `level_num` instead of `state["level_num"]`.